MergeCalendarsTogether.gs

```diff
@@ -258,131 +267,239 @@ function SortEvents(calendarId, items) {
       return
     }
     if (!DEBUG_ONLY) {
-      const result = new BatchRequest({
-        batchPath: 'batch/calendar/v3',
-        requests: calendarRequests,
-      });
+      const result = cal.executeCalls();
       if (!result.getResponseCode || result.getResponseCode() !== 200) {
         log.info(result)
       } else {
         log.debug('RESULT: ', result.toString(), '; ', result.getResponseCode() )
-        log.info(`${calendarRequests.length} events modified for ${calendarId}:`);
+        log.info(`${cal.apiCalls.length} events modified for ${cal.address}:`);
       }
     } else {
-      log.debug(`${calendarRequests.length} events would have been modified for ${calendarId}:`);
+      log.debug(`${cal.apiCalls.length} events would have been modified for ${cal.address}:`);
     }
-    const loggable = calendarRequests
-      .map(({method, endpoint, summary}) => ({method, endpoint, summary}))
-    log.info(`Requests for ${calendarId}`, JSON.stringify(loggable, null, 2));
+    const loggable = cal.apiCalls
+      .map(({method, endpoint, loggableSummary}) => ({method, endpoint, loggableSummary}))
+    log.info(`Requests for ${cal.address}`, JSON.stringify(loggable, null, 2));
   });
 }
 
+class MicrosoftCalendar {
+  constructor(obj) {
+    this.address = obj.address;
+    this.token = obj.token;
+    this.events;
+    this.apiCalls = [];
+  }
+
+  retrieve(startTime, endTime) {
+    const items = [];
+    let nextPage = MICROSOFT_ENDPOINT_BASE +
+      `/calendarview?startdatetime=${startTime.toISOString()}&enddatetime=${endTime.toISOString()}`
+    do {
+      const response = UrlFetchApp.fetch(nextPage, {
+        headers: {
+          'Authorization': 'Bearer ' + this.token,
+        }
+      });
+      const payload = JSON.parse(response.getContentText());
+      items.push(...payload.value.map(this.parseMicrosoftCal));
+      nextPage = payload["@odata.nextLink"];
+    } while(nextPage);
+    log.info(`Found ${items.length} items for ${this.address}`)
+    this.events = SortEvents(items);
+  }
+
+  parseMicrosoftCal(microsoftCal) {
+    return {
+      getId: () => microsoftCal.id,
+      start: microsoftCal.start,
+      end: microsoftCal.end,
+      description: microsoftCal.body.content,
+      location: microsoftCal.location.displayName,
+      summary: microsoftCal.subject,
+      transparency: microsoftCal.showAs === 'free' ? 'transparent' : 'opaque',
+      attendees: microsoftCal.attendees.map(({status, emailAddress}) => ({
+        email: emailAddress.address,
+        responseStatus: status.response,
+      })),
+    };
+  }
+
+  addCreateCall(event) {
+    const requestBody = {
+      subject: GetMergeSummary(event),
+      location: event.location,
+      reminderMinutesBeforeStart: 0,
+      body: {
+        contentType: 'text',
+        content: GetDesc(event),
+      },
+      start: event.start,
+      end: event.end,
+      attendees: GetAttendeeSelf(event, this.address),
```

Comment:
Ah, thanks for pointing this out - it doesn't look like the payload from MS provides any "self" indicator, but each attendee is a full object:
```
                    "type": "required",
                    "status": {
                        "response": "accepted",
                        "time": "0001-01-01T00:00:00Z"
                    },
                    "emailAddress": {
                        "name": "Jeremy Scherer",
                        "address": "first.last@example.com"
                    }
                },
```
so maybe a new class function instead of a "static" function since it'll be implementation dependent